core/src/smartcontracts/isi/domain.rs

```diff
@@ -35,18 +35,17 @@ pub mod isi {
                 .map_err(Error::Validate)?;
 
             wsv.modify_domain(&account_id.domain_id, |domain| {
-                match domain.accounts.entry(account_id.clone()) {
-                    Entry::Occupied(_) => Err(Error::Repetition(
+                if domain.get_account(&account_id).is_some() {
+                    return Err(Error::Repetition(
                         InstructionType::Register,
                         IdBox::AccountId(account_id.clone()),
-                    )),
-                    Entry::Vacant(entry) => {
-                        let _ = entry.insert(account.into());
-                        Ok(DomainEvent::Account(AccountEvent::Created(
-                            account_id.clone(),
-                        )))
-                    }
+                    ));
                 }
+
+                domain.add_account(account.into());
+                Ok(DomainEvent::Account(AccountEvent::Created(
+                    account_id.clone(),
+                )))
```

Comment:
I could add `contains_account` or some form of the entry API

Reply:
Wouldn't give us much and increase the surface area of the API.

Reply:
Yeah, I agree